### DISCUSSION TITLE
[Sequelize.Op.regexp]: '^[a-zA-Z0-9]{2}\d{10}$' not working with queryInterface.addConstraint sequelize

### DISCUSSION THREAD

**Discussion by AR98** [sequelize/sequelize#15436] [Help & Questions] [answered]
I need to add constraint(regex) on DB level so that no one able to enter/edit that particular column with invalid input on already existing table. I'm using Sequelize, Postgres.

I tried it using migration but still able to enter invalid value in DB. I was expecting that after running below migration I wouldn't able to enter/edit that column with wrong input but not working. Attaching my code below for you better understanding. 

`module.exports = {
  async up (queryInterface, Sequelize) {
     await queryInterface.addConstraint('Table',['column'], {
      type: 'check',
      name: 'checkNumber',
      where: {
       column: {
            [Sequelize.Op.regexp]: '^[a-zA-Z0-9]{2}\d{10}$'
        }
    }
    });
  },

  async down (queryInterface, Sequelize) {
     await queryInterface.removeConstraint('Table', 'checkNumber');
  }
};`

**ERROR: Fields must be specified through options.fields**

**Comment by ViniciusMarchi** [comment, marked as answer]
Hello, the error message displays that it is necessary to specify the `options.fields`, so I think that's the problem.
Try to set these fields and I believe this could be an solution. Considering your code make this change:
```javascript
  async up (queryInterface, Sequelize) {
    await queryInterface.addConstraint('Table', ['column'], {
      type: 'check',
      name: 'checkNumber',
      fields: ['column'], // insert this fields definition
      where: {
        column: {
          [Sequelize.Op.regexp]: '^[a-zA-Z0-9]{2}\d{10}$'
        }
      }
    });
  },

  async down (queryInterface, Sequelize) {
    await queryInterface.removeConstraint('Table', 'checkNumber');
  }
};
```
As you left only a snippet of code, I could not reproduce and test this solution, but it is in accordance with what is defined in the [Sequelize documentation](https://sequelize.org/api/v6/class/src/dialects/abstract/query-interface.js~queryinterface#instance-method-addConstraint). See if this solution helps you.

**Comment by LebonNic** [comment]
Hello,

I'm sorry to interfere in your discussion. I just wanted to know if it was possible to define a `CHECK` constraint at the `Model` level rather than in a migration file, so that the constraint is applied at `sequelize.sync({ force: true });` ?

Nicolas
